### PR TITLE
[nri-kubernetes] NRI kubernetes v3 upgrade shimming

### PR DESCRIPTION
**Pull request by kang-makes**
#### Is this a new chart

No

#### What this PR does / why we need it:

This is other approach that enables us to leave the compatibility layer separated from the code we really want to maintain.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

**Comment by kang-makes** [comment]
Rebase is needed but it is ready to start a discussion.

File: charts/nri-bundle/values.yaml [hunk 1]
```diff
@@ -2,10 +2,14 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
-# enables newrelic-infrastructure
+# Legacy nri-kubernetes chart, disabled by default
 infrastructure:
   enabled: true
```
Comment (roobre) [inline charts/nri-bundle/values.yaml:7]: I'd flip this to false, so new users don't have any hacks enabled by default.

File: charts/nri-bundle/values.yaml [hunk 2]
```diff
@@ -2,10 +2,14 @@
...
 
+# Latest nri-kubernetes chart
+nri-kubernetes:
+  enabled: false
```
Comment (roobre) [inline charts/nri-bundle/values.yaml:11]: And this enabled by default.

File: charts/nri-bundle/requirements.yaml [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
     version: 2.7.3
 
+  # New nri-kubernetes chart with compatibility layer.
+  - name: nri-kubernetes
+    alias: newrelic-infrastructure
+    repository: file://../nri-kubernetes
+    # Notes of why this is overengineered:
+    #  * This chart is still using Helm 2 so we cannot set various conditions like `infrastructure.enabled,newrelic-infrastructure.compatibilityLayer`
+    #  * Helm 2 also does not support `import-values` so we have to alias the new chart to the old one to get the values from it.
+    condition: newrelic-infrastructure.compatibilityLayer
```
Comment (roobre) [inline charts/nri-bundle/requirements.yaml:15]: If we don't set this to `false` (or `true`) in the values, helm will ignore the condition and render the chart.

Comment (kang-makes) [reply charts/nri-bundle/requirements.yaml:15]: Really good point. I would have to think about it.

**Comment by kang-makes** [comment]
Limitations of having Helm 2, a change in the name of the chart and a change in the values are too complex to a compatibility layer.

We cannot tackle all in the same release. What I tried with this PR is impossible.

Closing.

**Pull request closed without merging**